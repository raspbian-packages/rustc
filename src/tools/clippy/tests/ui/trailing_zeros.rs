#![feature(plugin, custom_attribute, stmt_expr_attributes)]
#![plugin(clippy)]
#![allow(unused_parens)]

fn main() {
    let x: i32 = 42;
    let _ = #[clippy(author)] (x & 0b1111 == 0);  // suggest trailing_zeros
    let _ = x & 0b1_1111 == 0; // suggest trailing_zeros
    let _ = x & 0b1_1010 == 0; // do not lint
}
