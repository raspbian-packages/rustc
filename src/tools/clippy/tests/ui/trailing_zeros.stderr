error: bit mask could be simplified with a call to `trailing_zeros`
 --> $DIR/trailing_zeros.rs:7:31
  |
7 |     let _ = #[clippy(author)] (x & 0b1111 == 0);  // suggest trailing_zeros
  |                               ^^^^^^^^^^^^^^^^^ help: try: `x.trailing_zeros() >= 4`
  |
  = note: `-D verbose-bit-mask` implied by `-D warnings`

error: bit mask could be simplified with a call to `trailing_zeros`
 --> $DIR/trailing_zeros.rs:8:13
  |
8 |     let _ = x & 0b1_1111 == 0; // suggest trailing_zeros
  |             ^^^^^^^^^^^^^^^^^ help: try: `x.trailing_zeros() >= 5`

error: aborting due to 2 previous errors

